torchaudio/functional.py

```diff
@@ -246,14 +247,22 @@ def spectrogram(
         # TODO add "with torch.no_grad():" back when JIT supports it
         waveform = torch.nn.functional.pad(waveform, (pad, pad), "constant")
 
+    # pack batch
+    shape = waveform.size()
+    waveform = waveform.reshape(-1, shape[-1])
+
     # default values are consistent with librosa.core.spectrum._spectrogram
     spec_f = _stft(
         waveform, n_fft, hop_length, win_length, window, True, "reflect", False, True
     )
 
+    # unpack batch
+    spec_f = spec_f.reshape(shape[:-1] + spec_f.shape[-3:])
```

Comment:
@vincentqb is support for batching going to be done optionally then? :grin: I can add it to the augmentation transforms if needed.

Reply:
Yeah, we decided to go ahead and add batching using reshape now :) we'll then update these codes when nested tensor comes out